power_grid_model_c/power_grid_model/include/power_grid_model/component/source.hpp

```diff
@@ -73,7 +73,26 @@ class Source : public Appliance {
         return changed;
     }
     // getter for u_ref for calc_param
-    template <bool sym> DoubleComplex calc_param() const { return u_ref_ * std::exp(1.0i * u_ref_angle_); }
+    DoubleComplex calc_param() const { return u_ref_ * std::exp(1.0i * u_ref_angle_); }
+    // This function receives the nominal voltage of the node and a min/max scaling enum
+    // and returns the reference voltage based on the voltage scaling factor c.
+    // the scaling factor is determined according to the IEC 60909 standard, which is shown in the table below:
+    //                   |   c_max   |   c_min   |
+    // Unom <= 1000V     |   1.10    |   0.95    |
+    // Unom > 1kV        |   1.10    |   1.00    |
+    // NOTE: for low voltage there is a difference in c for systems with a voltage tolerance of 6% or 10%.
+    // Here, a voltage tolerance of 10% is assumed.
+    DoubleComplex calc_param(std::pair<double, ShortCircuitVoltageScaling> const& data) const {
+        double voltage_scaling_c{1.0};
+        if (data.second == ShortCircuitVoltageScaling::minimum) {
+            if (data.first <= 1000.0) {
+                voltage_scaling_c = 0.95;
+            }
+        } else {
+            voltage_scaling_c = 1.10;
+        }
+        return voltage_scaling_c * std::exp(1.0i * u_ref_angle_);
```

Comment:
can you maybe make a separate (lambda) function of this? like

```suggestion
        double const voltage_scaling_c = [&data] {
            if (data.second == ShortCircuitVoltageScaling::maximum) {
                return 1.1;
            }
            if (data.first <= 1000.0) {
                return 0.95;
            }
            return 1.0;
        }();
        return voltage_scaling_c * std::exp(1.0i * u_ref_angle_);
```